File: 110E/Euclid.py
#!/usr/bin/env python3
import sys
########################################################
argv = [ 2166, 6099 ] # default values
# update argv with command line arguments
for j in range (1, len (sys.argv)):
  argv[ j -1] = int (sys.argv[ j])
m, n = argv
while True:
  r = m %n
  if r == 0: break
  m, n = n, r
print ("gcd({},{}) = {}".format (*argv, n))
########################################################
# log:

File: 110E/Euclid.pl
#! /usr/bin/env perl
my ($m, $n) = @ARGV;
#
if (not defined $m) {
    $m = 2166;
}
if (not defined $n) {
    $n = 6099;
}
my $s = "GCD($m,$n)="; # output string
#
if ($m < $n) { # E0
    $m, $n = $n, $m;
}
my $r = 0;
while (true) {
    $r = $m % $n;
    if ($r == 0) {
        last;
    }
    $m = $n;
    $n = $r;
}
print "$s$n\n";
# log:
